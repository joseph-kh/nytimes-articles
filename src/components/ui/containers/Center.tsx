import React from "react";
import { StyleSheet, View as ViewElement, ViewProps } from "react-native";

const Center: React.FC<ViewProps> = (props) => {
  const { style, children } = props;

  return (
    <ViewElement
      {...props}
      style={StyleSheet.flatten([styles.centerContainer, style])}
    >
      {children}
    </ViewElement>
  );
};

const styles = StyleSheet.create({
  centerContainer: {
    alignItems: "center",
    justifyContent: "center",
  },
});

export default Center;
