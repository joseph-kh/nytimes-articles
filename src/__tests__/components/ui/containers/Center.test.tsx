import React from "react";
import { create } from "react-test-renderer";
import Center from "@/components/ui/containers/Center";

describe("Center Component", () => {
  it("renders correctly", () => {
    const tree = create(<Center>Hello, World!</Center>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const customStyle = { backgroundColor: "red" };
    const tree = create(
      <Center style={customStyle}>Hello, World!</Center>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
