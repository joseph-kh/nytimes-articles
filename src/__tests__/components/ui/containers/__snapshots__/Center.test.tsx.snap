// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component applies custom styles 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "justifyContent": "center",
    }
  }
>
  Hello, World!
</View>
`;

exports[`Center Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  Hello, World!
</View>
`;
